engine/runtime-compiler/src/main/java/org/enso/compiler/pass/ChainedMiniPass.java

```diff
@@ -15,37 +15,43 @@ private ChainedMiniPass(MiniIRPass firstPass, MiniIRPass secondPass) {
   }
 
   static MiniIRPass chain(MiniIRPass firstPass, MiniIRPass secondPass) {
-    if (firstPass == null) {
-      return secondPass;
-    }
     return new ChainedMiniPass(firstPass, secondPass);
   }
 
   @Override
   public MiniIRPass prepare(IR parent, Expression current) {
-    var first = firstPass.prepare(parent, current);
-    var second = secondPass.prepare(parent, current);
-    if (first == firstPass && second == secondPass) {
+    var firstPrepared = firstPass == null ? null : firstPass.prepare(parent, current);
+    var secondPrepared = secondPass == null ? null : secondPass.prepare(parent, current);
+    if (firstPrepared == firstPass && secondPrepared == secondPass) {
       return this;
     } else {
-      return new ChainedMiniPass(first, second);
+      return new ChainedMiniPass(firstPrepared, secondPrepared);
     }
   }
 
   @Override
   public Expression transformExpression(Expression ir) {
-    var fstIr = firstPass.transformExpression(ir);
-    var sndIr = secondPass.transformExpression(fstIr);
+    var fstIr = firstPass == null ? ir : firstPass.transformExpression(ir);
+    var sndIr = secondPass == null ? fstIr : secondPass.transformExpression(fstIr);
     return sndIr;
   }
 
+  @Override
+  public Module transformModule(Module moduleIr) {
+    var first = firstPass == null ? moduleIr : firstPass.transformModule(moduleIr);
```

Comment:
For consistency with `transformExpression` method call the variable `fstIr`, maybe?

Reply:
Reasonable renaming is always a good idea. Done in b3f750eb52704a386b26400f54cc8d685ff3bc60